plugins/cpp/parser/src/cppparser.cpp

```diff
@@ -285,11 +292,25 @@ void CppParser::worker()
 
     clang::tooling::ClangTool tool(*compilationDb, command.Filename);
 
-    tool.run(&factory);
+    int error = tool.run(&factory);
+
+    model::FilePtr file = _ctx.srcMgr.getFile(command.Filename);
+    if (error)
+    {
+      file->parseStatus = model::File::PSPartiallyParsed;
+
+      LOG(error)
+        << '(' << index << '/' << _compileCommands.size() << ')'
+        << " Parsing " << command.Filename << " has been failed.";
+    }
+    else
+      file->parseStatus = model::File::PSFullyParsed;
+
+    _ctx.srcMgr.updateFile(*file);
```

Comment:
Do we really update the file status here? Because that's why `error_` parameter is introduced in `addCompileCommand` function.